src/pages/index.js

```diff
@@ -103,41 +140,137 @@ function closeModalOverlay(event) {
 
 function handleEditFormSubmit(event) {
   event.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal, settings);
+
+  const submitButton = event.submitter;
+  submitButton.textContent = "Saving...";
+  setButtonText(submitButton, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = editModalNameInput.value;
+      profileDescription.textContent = editModalDescriptionInput.value;
+      closeModal(editModal, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+      submitButton.textContent = "Save";
+    });
+}
+
+function setButtonText(
+  button,
+  isLoading,
+  defaultText = "Save",
+  loadingText = "Saving..."
+) {
+  if (isLoading) {
+    button.textContent = loadingText;
+  } else {
+    button.textContent = defaultText;
+  }
 }
 
 function handleAddCardSubmit(event) {
   event.preventDefault();
   const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  event.target.reset();
-  disabledButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+  setButtonText(cardSubmitButton, true);
+
+  api
+    .addNewCards(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      event.target.reset();
+      disabledButton(cardSubmitButton, settings);
+      closeModal(cardModal);
+    })
+    .catch((error) => console.error(error))
+    .finally(() => {
+      setButtonText(cardSubmitButton, false);
+    });
+}
+
+function handleavatarFormSubmit(event) {
+  event.preventDefault();
+  setButtonText(cardSubmitButton, true);
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      console.log("Success:", data);
+      closeModal(avatarModal);
+      event.target.reset();
+      disabledButton(avatarSubmitButton, settings);
+    })
+    .catch((error) => console.error("Error:", error));
+  setButtonText(cardSubmitButton, false);
+}
+
+function handleDeleteSubmit(event) {
+  event.preventDefault();
+  setButtonText(cardSubmitButton, true);
```

Comment:
`cardSubmitButton` is for the card form, but here should be `const submitButton = event.submitter;`


```js
function handleDeleteSubmit(event) {
  event.preventDefault();
  const submitButton = event.submitter;  // get the submit button universally

  setButtonText(submitButton, true, "Delete", "Deleting...");  // use it, reassigning the texts
  api
    .deleteCard(selectedCardId)
    .then(() => {
      selectedCard.remove();
      closeModal(deleteModal);
    })
    .catch(console.error)
    .finally(() => {
      setButtonText(submitButton, false, "Delete", "Deleting..."); // use it, reassigning the texts
    });
}

```

Reply:
* When I delete a card, the button text should display `Deleting...`